modules/apps/captcha/captcha-taglib/src/main/java/com/liferay/captcha/taglib/servlet/taglib/CaptchaTag.java

```diff
@@ -70,14 +75,31 @@ private String _getURL(HttpServletRequest httpServletRequest) {
 			return _url;
 		}
 
+		String portletId = PortalUtil.getPortletId(httpServletRequest);
+
+		try {
+			CaptchaConfiguration captchaConfiguration =
+				ConfigurationProviderUtil.getSystemConfiguration(
+					CaptchaConfiguration.class);
+
+			if (captchaConfiguration.enableSimpleCaptchaHeadlessAPI() &&
+				Validator.isNotNull(portletId)) {
+
+				return "/o/captcha/v1.0/simple?portletId=" + portletId;
+			}
```

Comment:
API will not be used if the portletId is null? I don't think so. Please help check. Thanks.

Reply:
Fixed it. Thank